transaction.go

```diff
@@ -30,14 +30,35 @@ import (
 
 // A Transaction is a full transaction object containing a payload and signatures.
 type Transaction struct {
-	Script             []byte
-	Arguments          [][]byte
-	ReferenceBlockID   Identifier
-	GasLimit           uint64
-	ProposalKey        ProposalKey
-	Payer              Address
-	Authorizers        []Address
-	PayloadSignatures  []TransactionSignature
+	// Script is the transaction script as UTF-8 encoded Cadence source code
+	Script []byte
+
+	// Arguments that are passed to the Cadence transaction
+	Arguments [][]byte
+
+	// ReferenceBlockID is a reference to another.
+	//
+	// A transaction is expired after specific number of blocks (defined by network) counting from this block.
+	// For example, if block reference is pointing to a block with height of X and network limit is 10,
+	// a block with x+10 height is the last block that is allowed to include this transaction.
+	// This reference can be set to an older block if the tx should expire faster.
+	ReferenceBlockID Identifier
+
+	// GasLimit is the max amount of computation which is allowed to be used during this transaction
+	GasLimit uint64
+
+	// ProposalKey is key of the account used to propose the transaction
+	ProposalKey ProposalKey
+
+	// Payer is the account that pays the feed for this transaction
+	Payer Address
+
+	Authorizers []Address
+
+	// PayloadSignatures is the list of account signatures excluding signature of the payer account
+	PayloadSignatures []TransactionSignature
+
+	// EnvelopeSignatures is the payer signature over the envelope (payload + payload signatures)
```

Comment:
Could you clarify this a bit? Should it be `payer + payload` signatures?

Reply:
Sorry, I just copied this over from the flow-go codebase

Reply:
I think I understand what it means now. I thought it was saying (payload signature + payload signature) and I thought it was a typo. But is actually saying (the entire payload + the signature for the payload).